src/Features/Core/Portable/CodeLens/CodeLenseReferenceService.cs

```diff
@@ -0,0 +1,269 @@
+            return reference.Definition.Locations.Intersect(
+                queriedDefinition.Locations, LocationComparer.Instance).Any();
+        }
+
+        private static async Task<IEnumerable<Tuple<ReferencedSymbol, bool>>> GetReferences(ISymbol symbol, Solution solution, CancellationToken cancellationToken)
+        {
+            var references = await SymbolFinder.FindReferencesAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
+
+            // Exclude the following kind of symbols:
+            //  (a) Implicitly declared symbols (such as implicit fields backing properties)
+            //  (b) Symbols that can't be referenced by name (such as property getters and setters).
+            //  (c) Metadata only symbols, i.e. symbols with no location in source.
+            IEnumerable<ReferencedSymbol> filteredReferences = from reference in references
+                                                               let symbolDef = reference.Definition
+                                                               where FilteringHelpers.FilterReference(symbolDef, reference)
+                                                               select reference;
+
+            return filteredReferences.Select(reference => Tuple.Create(reference, IsQueriedDefinition(reference, symbol)));
+        }
+
+        private static async Task<Tuple<SyntaxNode, IEnumerable<Tuple<ReferencedSymbol, bool>>>> FindReferencesAsync(Solution solution, Document document, SyntaxNode syntaxNode, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            var symbol = semanticModel.GetDeclaredSymbol(syntaxNode, cancellationToken);
+            if (symbol == null)
+            {
+                return null;
+            }
+
+            var references = await SymbolFinder.FindReferencesAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
+
+            // Exclude the following kind of symbols:
+            //  (a) Implicitly declared symbols (such as implicit fields backing properties)
+            //  (b) Symbols that can't be referenced by name (such as property getters and setters).
+            //  (c) Metadata only symbols, i.e. symbols with no location in source.
+            IEnumerable<ReferencedSymbol> filteredReferences = from reference in references
+                                                               let symbolDef = reference.Definition
+                                                               where FilteringHelpers.FilterReference(symbolDef, reference)
+                                                               select reference;
+
+            var allReferences =
+                filteredReferences.Select(reference => Tuple.Create(reference, IsQueriedDefinition(reference, symbol)));
+
+            // Search through all other projects to see if we can a matching symbol in other projects
+            foreach (var additionalProject in document.Project.Solution.Projects)
```

Comment:
I think jasonmal added this code before dpoeschl did the work for linked files at the workspace api layer. I _think_ this could be removed.